### PR TITLE
No hyphen for GUIDs during serialization

### PR DESCRIPTION
**Pull request by QilongTang**
Please Note:
1. Before submitting the PR, please review [How to Contribute to Dynamo](https://github.com/DynamoDS/Dynamo/blob/master/CONTRIBUTING.md)
2. Dynamo Team will meet 1x a month to review PRs found on Github (Issues will be handled separately)
3. PRs will be reviewed from oldest to newest
4. If a reviewed PR requires changes by the owner, the owner of the PR has 30 days to respond. If the PR has seen no activity by the next session, it will be either closed by the team or depending on its utility will be taken over by someone on the team
5. PRs should use either Dynamo's default PR template or [one of these other template options](https://github.com/DynamoDS/Dynamo/wiki/Choosing-a-Pull-Request-Template) in order to be considered for review.
6. PRs that do not have one of the Dynamo PR templates completely filled out with all declarations satisfied will not be reviewed by the Dynamo team.
7. PRs made to the `DynamoRevit` repo will need to be cherry-picked into all the DynamoRevit Release branches that Dynamo supports. Contributors will be responsible for cherry-picking their reviewed commits to the other branches after a `LGTM` label is added to the PR.

### Purpose

[QNTM-1123](https://jira.autodesk.com/browse/QNTM-1123)  As a Dynamo Dev, I want Dynamo Id json serialized with no dashes

Internally, we are still using Default GUID formats with hyphens. The GUID constructor is tolerant enough. But to be consistent with AVP and CoGs in terms of Id serialization, we plan to unify all the serialization in 32 digits string format without hyphens. There will be another PR in AVP to change Guids serialization as well.

This PR:
1. Use GUIDs without hyphens for Node, Notes, Connectors, Groups, Ports
2. Modify Serialization Tests
3. Works together with the AVP part PR: https://github.com/DynamoDS/AutodeskVisualProgramming/pull/271

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [x] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions), and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers
@mjkkirschner 

### FYIs
@ColinDayOrg @alfarok @smangarole

### CODE REVIEW

**Comment by mjkkirschner** [comment]
Note that it's currently safe in CoGS for UUID of workspace to have hyphens.

File: src/DynamoCore/Graph/Nodes/NodeModel.cs [hunk 1]
```diff
@@ -722,7 +722,7 @@ public string AstIdentifierGuid
         {
             get
             {
-                return GUID.ToString().Replace("-", string.Empty).ToLower();
+                return GUID.ToString("N");
```
Comment (mjkkirschner) [inline src/DynamoCore/Graph/Nodes/NodeModel.cs]: what does the "N" do, does it also accomplish the toLower field? Why are you changing this AST id - we don't serialize this I don't think, it should only be used internally for the VM...

Comment (QilongTang) [inline src/DynamoCore/Graph/Nodes/NodeModel.cs]: The default option is "D" if no Parma passed in which is the format with dashes. "N" is the one for format without dashes and in lowercase. This is doing what that code does so I just replaced the these code.

**Comment by QilongTang** [comment]
Ok. Then I will remove the second Todo since it's not necessary.

**Comment by mjkkirschner** [comment]
LGTM

**Pull request merged**